packages/entities/src/lib/update.mutation.ts

```diff
@@ -47,24 +78,51 @@ export function updateEntities<
   updater: U,
   options: BaseEntityOptions<Ref> = {}
 ): Reducer<S> {
-  return function (state) {
-    const { ref: { entitiesKey } = defaultEntitiesRef } = options;
+  return function (state, context) {
+    const { ref = defaultEntitiesRef } = options;
     const updatedEntities = {} as Record<
       getIdType<S, Ref>,
       getEntityType<S, Ref>
     >;
 
+    const idProp = getIdKey<keyof getEntityType<S, Ref>>(context, ref);
```

Comment:
I prefer to go with my first suggestion and pass explicitly this option. There is no need to allocate memory and do all of this if there is no need to.

Add an if statement in line 63 at the original code that takes the updated state and call an updateIds function

Reply:
This isn't a common use case, and I don't want it to affect in any way the original code.

Reply:
I have a better idea

Reply:
We can create a specific mutation:

```ts
store.update(
  updateEntityId(oldId, newId),
  updateEntities(newId, updater)
)
```

Reply:
This will make it tree-shakable and cleaner

Reply:
If I understand correctly, the purpose of the proposed flag is to get rid of the `changedIds`, which determines if id was changed. Since this use-case is addressing the "optimistic updates" use-case, the `changedIds` will contain a single item 99% of the time. The memory consumption of that object will be minimal.

Besides, even if the flag is introduced, we still need to track what was the old and the new id for each updated entity. I'm not sure how introducing a flag will let us get around that.

Reply:
Never mind. I prefer to go with a new mutation function:

```ts
store.update(
  updateEntitiesIds(oldId, newId), // support single and array
  updateEntities(newId, updater)
)
```

Reply:
interesting. Looks clean! And should be much simpler to implement too! I'm on it